python/grass/app/runtime.py

```diff
@@ -29,15 +29,34 @@ def get_grass_config_dir(major_version, minor_version, env):
     Determines path of GRASS GIS user configuration directory.
     """
     # The code is in sync with grass.app.runtime (but not the same).
+    env_dirname = "APPDATA" if WINDOWS else "HOME"
+    if env.get(env_dirname) is None:
+        raise KeyError(
+            f"The {env_dirname} variable is not set, ask your operating"
+            " system support"
+        )
+    if not os.path.exists(env.get(env_dirname)):
+        raise IsADirectoryError(
```

Comment:
Raising [IsADirectoryError](https://docs.python.org/3/library/exceptions.html) when it is confusing:

> exception IsADirectoryError
>
> Raised when a file operation (such as [os.remove()](https://docs.python.org/3/library/os.html#os.remove)) is requested on a directory. Corresponds to errno [EISDIR](https://docs.python.org/3/library/errno.html#errno.EISDIR).

[NotADirectoryError](https://docs.python.org/3/library/exceptions.html#NotADirectoryError) seems to be appropriate, but in that case, the test should be [isdir](https://docs.python.org/3/library/os.path.html#os.path.isdir) not [exists](https://docs.python.org/3/library/os.path.html#os.path.exists).

Reply:
Changed in 1ea52eb40f.